src/Evolution/Systems/CurvedScalarWave/Worldtube/ElementActions/SendToWorldtube.hpp

```diff
@@ -161,8 +162,26 @@ struct SendToWorldtube {
     Parallel::receive_data<Worldtube::Tags::SphericalHarmonicsInbox<Dim>>(
         worldtube_component, db::get<::Tags::TimeStepId>(box),
         std::make_pair(element_id, std::move(Ylm_coefs)));
+    if (db::get<Tags::CurrentIteration>(box) <
+        db::get<Tags::MaxIterations>(box) - 1) {
```

Comment:
Should this logic just go in `IteratePunctureField` instead? I think it makes more sense for `IteratePunctureField` to be the one to increase the `current_iteration` and determine if it actually needs to compute another iteration of the puncture field or go to BCs

Reply:
That's what I had originally but I think this way is better because it automatically handles the cases where `MaxIterations` is equal to 0 (geodesic) or 1 and the `IteratePunctureField` and `IterateAccelerationTerms` actions are skipped entirely.

Reply:
Ahh I see. I guess I was a bit confused because checking the iterations didn't seem related to sending data to the worldtube. But the beginning of the iterate actions can determine if an iteration needs to happen or not. I don't have a strong opinion on this so I'll defer to you.